src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs

```diff
@@ -8,10 +8,10 @@ namespace Microsoft.Azure.ServiceBus.Amqp
     using Azure.Amqp;
     using Core;
     using Filters;
-    using Messaging.Amqp;
 
     internal sealed class AmqpSubscriptionClient : IInnerSubscriptionClient
```

Comment:
>internal sealed [](start = 4, length = 15)

I think we should try to add a new sample which clearly mentions that its for advanced scenarios and use MessageReceivers and sessions etc just to validate. 
You could take it up as a separate PR, but it will help you close on few things like whether this class should be internal/public, or whether the class needs to be renamed